src/Raven.Server/Utils/UnhandledExceptions.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Threading.Tasks;
+using Raven.Client.Extensions;
+using Sparrow.Logging;
+
+namespace Raven.Server.Utils
+{
+    public class UnhandledExceptions
+    {
+        private static readonly TimeSpan TimeToWaitForLog = TimeSpan.FromSeconds(15);
+
+        public static void Track(Logger logger)
+        {
+            AppDomain.CurrentDomain.UnhandledException += (sender, args) =>
+            {
```

Comment:
This is going to kill the process, it ALWAYS interests us.
That said, are we sure that this actually give us a good error _when running as a service_?

Reply:
We are re-checking that, just in case.